Cleanup of summary related interfaces

vladsud opened the pull request:
1. Better typing for getGitType()
2. Removal of summary commits
3. Better typing for ODSP interfaces
4. removal of duplicated code

vladsud committed:
1. Better typing for getGitType()
2. Removal of summary commits
3. Better typings for ODSP interfaces
4. removal of duplicated code

arinwt commented on server/routerlicious/packages/protocol-base/src/blobs.ts:
```diff
@@ -42,14 +41,12 @@ export function getGitMode(value: SummaryObject): string {
  * @returns the type of summary object
  */
-export function getGitType(value: SummaryObject): string {
+export function getGitType(value: SummaryObject): "blob" | "commit" | "tree" | "attachment" {
```
In the function definition you have removed the possibility to return "commit", but the return type signature of the function includes "commit".

arinwt commented on packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:
```diff
@@ -1106,18 +1110,19 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                 }
             }
 
+            const type = getGitType(summaryObject) as "blob" | "commit" | "tree" | "attachment";
```
May want to put a note somewhere that these casts can be removed once prerelease of server is published.  Are you planning to do in a follow-up PR?

vladsud replied:
Opened https://github.com/microsoft/FluidFramework/issues/4835 to follow up

arinwt reviewed (approved):
Happy to see commit removed, just to verify my understanding is the following true?

The idea is to remove commit concept from write operations/types, but leave it in the read code/types (for back-compat).

arinwt commented on server/routerlicious/packages/protocol-base/src/blobs.ts:
```diff
@@ -42,14 +41,12 @@ export function getGitMode(value: SummaryObject): string {
     switch (type) {
         case SummaryType.Blob:
             return "blob";
-        case SummaryType.Commit:
```
The function name `getGitType` and location in the code makes me feel like it should be possible to return "commit", since it is scoped to git functionality vs. the specific routerlicious usage of git objects.. if the latter is the case, then maybe at least a brief comment could clarify that this is scoped to routerlicious git objects only.

vladsud replied:
Not sure I follow you. The return type is a function of input. I.e. if SummaryObject had commits, then getGitType would return commits. So it's more a function of summaries than git/r11s

vladsud committed:
Merge branch 'main' of https://github.com/Microsoft/FluidFramework into Summary_Commit

vladsud committed:
PR feedback, small changes

vladsud committed:
Compile, better default handling of non-reachable cases

vladsud committed:
Make it build

vladsud committed:
Add missing cases when compiling client & server bits separately

anthony-murphy commented on packages/drivers/odsp-driver/src/createFile.ts:
```diff
@@ -165,7 +167,7 @@ export function convertSummaryToSnapshotTreeForCreateNew(summary: ISummaryTree):
         const entry: SnapshotTreeEntry = {
             path: encodeURIComponent(key),
-            type: getGitType(summaryObject),
+            type: getGitType(summaryObject) as "blob" | "tree" | "blob",
```
getGitType returns blob tree or attachment, but you are casting away attachment here. you do a similar cast in a couple places. i don't quite understand why it's ok for these types not to match

vladsud replied:
Casting was here as an attempt to allow server & client changes to go together without need to have multiple commits, but that did not work out.
That said, it was wrong - getGitType() should never return "attachment", it should return "blob" for it same way as extra copy of that API in r11s was doing it - move that to protocol-base and changes this place and another place where we were special-casing "attachment" case.

vladsud committed:
Fix "attachment" case

vladsud committed:
Merge branch 'main' of https://github.com/Microsoft/FluidFramework into Summary_Commit

# Conflicts:
#	packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts
#	packages/drivers/routerlicious-driver/src/documentStorageService.ts

vladsud committed:
Make it compile

vladsud commented:
> Happy to see commit removed, just to verify my understanding is the following true?
> The idea is to remove commit concept from write operations/types, but leave it in the read code/types (for back-compat).

I do not think we have code anywhere except in ODSP driver that deals with commits. I.e. once we switched to summaries, I believe we removed all the code that deal with commits even for load. Searching for ".commits" mostly confirms it - drivers have some references, r11s has some, but I see nothing in runtime.
